src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationAndGeneratorDriverTranslationAction_Actions.cs

```diff
@@ -113,6 +118,9 @@ public override async Task<Compilation> TransformCompilationAsync(Compilation ol
 
                     return oldCompilation.RemoveAllSyntaxTrees().AddSyntaxTrees(syntaxTrees);
                 }
+
+                // Because this removes all trees, it'd also remove the generated trees
+                public override bool CanApplyToCompilationWithGeneratedDocuments => false;
```

Comment:
so what happens in this case.  we remove all the trees, tehn teh generator runs and produces the trees again?

Reply:
We're not going to attempt to reuse a prior compilation; the generator will run, produce new output, and we'll reuse what we can. Practically this operation is mostly done when we're changing parse options, so there's not much we can reuse.